R/helpers_knitr_GeneralModel.R

```diff
@@ -941,3 +962,201 @@ knit_print.LogisticLogNormalOrdinal <- function(
+  # knit_print.ModelParamsNormal expects no row or column names
+  rownames(posterior@cov) <- NULL
+  colnames(posterior@cov) <- NULL
+
+  rv <- paste0(
+    "A logistic log normal model will describe the relationship between dose and toxicity: ",
+    "$$ p(Tox | d) = f(X = 1 | \\theta, d) = \\frac{", z, "}{1 + ", z, "} $$\\n ",
+    "The prior is expressed in terms of pseudo data and, consequently, the number ",
+    " of cases and of ",
+    tox_label,
+    " need not be whole numbers.\n\nThe pseudo data are ",
+    "defined in the following table:\n\n",
+    paste0(
+      do.call(
+        function(x) {
+          kableExtra::kable_styling(
+            knitr::kable(x, col.names = c("Dose", "N", tox_label)),
+            bootstrap_options = c("striped", "hover", "condensed")
+          )
+        },
+        list(x = y$pseudoData)
+      ),
+      collapse = "\n"
+    ),
+    ifelse(
+      nrow(y$data) == 0,
+      "\n\nNo observed data has yet been recorded.\n",
+      paste(
+        "\n\nThe observed data are given in the following table:\n\n",
+        paste((do.call(knitr::kable, list(x = y$data))), collapse = "\n")
+      )
+    ),
+    knit_print(
+      posterior,
+      preamble = paste0(
+        "\n\nTogether, the pseudo and observed data give rise to ",
+        "the following posterior for the model parameters:\n\n"
+      ),
+      params = c("\\phi_1" = "phi1", "\\phi_2" = "phi2"),
+      theta = "\\phi",
+      asis = FALSE,
+      ...
+    ),
+    "\n\n"
+  )
+
+  if (asis) {
+    rv <- knitr::asis_output(rv)
+  }
+  rv
+}
+
+
+# Effloglog ----
+
+#' @description `r lifecycle::badge("experimental")`
+#' @param eff_label (`character`)\cr the term used to describe efficacy
+#' @rdname knit_print
+#' @export
+#' @method knit_print Effloglog
+knit_print.Effloglog <- function(
+    x,
+    ...,
+    use_values = TRUE,
+    fmt = "%5.2f",
+    params = NA,
+    tox_label = "DLAE",
+    eff_label = "efficacy",
+    label = "participant",
+    preamble = "The prior for &theta; is given by\\n",
+    asis = TRUE) {
+  assert_flag(asis)
+  assert_character(eff_label, len = 1, any.missing = FALSE)
+
+  # Prepare
+  if (length(tox_label == 1)) {
+    tox_label <- c(tox_label, paste0(tox_label, "s"))
+  }
+  if (length(eff_label == 1)) {
+    eff_label <- c(eff_label, paste0(eff_label, "s"))
+  }
+  if (length(label == 1)) {
+    label <- c(label, paste0(label, "s"))
+  }
+
+
+  y <- tidy(x)
+  # knit_print.ModelParamsNormal expects no row or column names
+  posterior <- ModelParamsNormal(mean = c(x@theta1, x@theta2), cov = x@Q)
+  rownames(posterior@cov) <- NULL
+  colnames(posterior@cov) <- NULL
+
+  rv <- paste0(
+    "A linear log-log model with a pseudo data prior will describe the ",
+    "relationship between dose and ",
+    eff_label[1],
+    ".  The model is given by\n ",
+    "$$ y_i = \\theta_1 + \\theta_2 \\cdot \\log(\\log(d_i + k)) + \\epsilon_i $$\\n ",
```

Comment:
I would consistently use `w_i` for efficacy, as that aligns with the corresponding `Data` slots and the table. Maybe not just here but also elsewhere

Reply:
I used y<sub>i</sub> to align with the online documentation.  Would you like me to update the docs as well?

Reply:
ah, ok. yes that would be good.